crates/iceberg/src/spec/partition.rs

```diff
@@ -54,6 +56,24 @@ impl PartitionSpec {
     pub fn builder() -> PartitionSpecBuilder {
         PartitionSpecBuilder::default()
     }
+
+    /// Construct the partition type from schema
+    pub fn partition_type(&self, schema: &Schema) -> Result<StructType, Error> {
+        let mut fields = Vec::with_capacity(self.fields.len());
+        for field in &self.fields {
+            let field = schema.field_by_id(field.source_id).ok_or_else(|| {
```

Comment:
I think this is incorrect? The partition type should be determined by type after transformation.

Reply:
Yes. My bad.🥵